### PR TITLE
feat: curl trace and prove them locally

### PR DESCRIPTION
**Pull request by Lawliet-Chan**

### CODE REVIEW

File: curl_prove.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+mkdir -p all_traces
+
+i=0
+for hash in `jq .[].header_hash l2_blocks.json`;
+do
+	echo "-------- Downloading $i $hash --------"
+	curl --location --request POST 'https://prealpha.scroll.io/l2' \
+	--header 'Content-Type: application/json' --data-raw '{"jsonrpc": "2.0","method": "eth_getBlockResultByHash","params":['${hash}'],"id": 1}'  > ./all_traces/${i}.trace
+	echo "-------- Proving $i $hash --------"
+	./target/release/prove --agg ./all_traces/${i}.proof --params zkevm/test_params --seed zkevm/test_seed
```
Comment (HAOYUatHZ) [inline curl_prove.sh:17]: missing trace?

File: bin/src/prove.rs [hunk 2]
```diff
@@ -62,4 +68,12 @@ fn main() {
         let mut f = File::create(path).unwrap();
         f.write_all(state_proof.proof.as_slice()).unwrap();
     }
+
+    if let Some(path) = args.agg_proof_path {
+        let agg_proof = prover
+            .create_agg_circuit_proof(&trace)
+            .expect("cannot generate evm_proof");
```
Comment (HAOYUatHZ) [inline bin/src/prove.rs:75]:
```suggestion
            .expect("cannot generate agg proof");
```

**Comment by lispc** [comment]
Looks good now ( But later we will have more blocks so if we can reuse prover instance within one single process we can avoid generate poving key again and again and save a lot of testing time. We can do it later)

**Comment by Lawliet-Chan** [comment]
> Looks good now ( But later we will have more blocks so if we can reuse prover instance within one single process we can avoid generate poving key again and again and save a lot of testing time. We can do it later)

Get your point. I will do it.

**Comment by HAOYUatHZ** [comment]
closed in favor of https://github.com/scroll-tech/common-rs/pull/46

**Pull request closed without merging**